compaction/compaction_manager.cc

```diff
@@ -1180,7 +1180,7 @@ void compaction_manager::submit(table_state& t) {
 
     // OK to drop future.
     // waited via task->stop()
-    (void)perform_compaction<regular_compaction_task_executor>(tasks::task_info{}, t).discard_result();
+    (void)perform_compaction<regular_compaction_task_executor>(tasks::task_info{}, t).then_wrapped([] (auto f) { f.ignore_ready_future(); });
```

Comment:
@Deexie hello Aleksandra, it seems `task` was removed and was superseded by `compaction_task_executor`. could you help update it to reflect the latest code? is it `task_manager::task::done()` ?

Reply:
Done in #15023